cmd/consumer/main.go

```diff
@@ -143,7 +143,7 @@ func appendToFile(filename, text string) error {
 func (rp *RecordProcessor) Shutdown(checkpointer kcl.Checkpointer, reason string) error {
 	if reason == "TERMINATE" {
 		fmt.Fprintf(os.Stderr, "Was told to terminate, will attempt to checkpoint.\n")
-		rp.firehoseWriter.FlushAll()
+		rp.firehoseWriter.Shutdown()
 		rp.checkpoint(checkpointer, "", 0)
```

Comment:
1. `Flush([][]byte)` is part of the message batcher interface, so i was trying to separate it from that. But `FlushAll` does make more sense I think. I'll change back.
2. I actually don't know-- this is unchanged from the `SampleRecordProcessor`. I'll look into it.